### PR TITLE
[K32W0] Fix OTA memory allocation

### PR DESCRIPTION
**Pull request by doru91**
#### Problem
* See https://github.com/project-chip/connectedhomeip/issues/13393

#### Change overview
Use the non-buggy version from ESP32.

#### Testing
How was this tested? (at least one bullet point required)
* not tested yet, still working on OTA end-to-end flow

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]
Fast tracking, given platform changes by a platform maintainer.

File: src/platform/nxp/k32w/k32w0/OTAImageProcessorImpl.cpp [hunk 1]
```diff
@@ -142,29 +142,30 @@ void OTAImageProcessorImpl::HandleProcessBlock(intptr_t context)
 
 CHIP_ERROR OTAImageProcessorImpl::SetBlock(ByteSpan & block)
 {
-    if ((block.data() == nullptr) || block.empty())
+    if (!IsSpanUsable(block))
     {
+        ReleaseBlock();
```
Comment (bzbarsky-apple) [inline src/platform/nxp/k32w/k32w0/OTAImageProcessorImpl.cpp:147]: Why this release?

Comment (carol-apple) [reply src/platform/nxp/k32w/k32w0/OTAImageProcessorImpl.cpp:147]: Agreed, there would be nothing to release

**Review by selissia — APPROVED** [review]

File: src/platform/nxp/k32w/k32w0/OTAImageProcessorImpl.cpp (hunk 1 above)
Comment (doru91) [reply src/platform/nxp/k32w/k32w0/OTAImageProcessorImpl.cpp]: agree, removed the unneeded release.

**Pull request merged**